python/tvm/contrib/graph_runtime.py

```diff
@@ -162,7 +162,11 @@ def set_input(self, key=None, value=None, **params):
             keys = list(params.keys())
             keys.sort(key=lambda x: -np.prod(params[x].shape))
             for k in keys:
-                self._get_input(k).copyfrom(params[k])
+                # TODO(zhiics) Skip the weights for submodule in a better way.
+                # We could get all inputs required by graphruntime first,
+                # we should use MetadataModule for initialization.
```

Comment:
This part of the logic seems to be a bit too hacky, cryptic Shall we simply skip inside the set_params instead? If the weights are emedded in the metadata, do we still allow overriding of weights?

Reply:
Yeah, we can do some checking at the C++ side as well. We will need to change graph runtime later to let it have a "__init" function so that we can initialize it with params. And we will need to remove some of the current ways to set the params up. For example, the `set_input`  API should not take `**params` and the chunk of hacky code will be removed. That needs some more thinking and may worth an RFC.

Reply:
Perhaps it is possible to automatically call the `__init` function in module construction and binary loading time ?

Reply:
yeah, that is something what I was thinking of as well. Let me spend some time to think about some possible ways. I will post an RFC with some plans because graph runtime affects apparently every downstream users.